### ISSUE TITLE
Liberty startup issue:  NumberFormatException in com.ibm.ws.kernel.service.util.JavaInfo when running on GraalVM 1.0 RC16

### ISSUE THREAD

**Issue by MaartenSmeets** [OpenLiberty/open-liberty#7372]
When running JAR generated with mvn clean package from https://github.com/MaartenSmeets/jvmperformance/tree/master/microservicetestcode/MicroProfile
 (using net.wasdev.wlp.maven.parent:liberty-maven-app-parent:2.6.4,
net.wasdev.wlp.maven.parent:liberty-maven-plugin:2.6.4,
io.openliberty:openliberty-runtime:19.0.0.4)

on GraalVM 1.0 RC16 using the following Dockerfile

```
FROM oracle/graalvm-ce:1.0.0-rc16
VOLUME /tmp
ARG JAR_FILE
COPY ${JAR_FILE} app.jar
ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-XX:+UnlockExperimentalVMOptions","-Xmx2G","-Xms2G","-jar","/app.jar"]
```

with
```
docker build -t testimage -f Dockerfile --build-arg JAR_FILE=target/mp-rest-service.jar .
docker run -d -p 8080:8080 testimage
```

My application will not start and I get the following error (docker logs testimage).

```
[ERROR   ] CWWKE0701E: FrameworkEvent ERROR org.osgi.framework.BundleException: Exception in com.ibm.ws.config.internal.WSConfigActivator.start() of bundle com.ibm.ws.config.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:835)
	at [internal classes]
Caused by: java.lang.ExceptionInInitializerError
	at com.ibm.ws.config.admin.internal.UpdateQueue.<init>(UpdateQueue.java:152)
	... 1 more
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.ibm.ws.kernel.service.util.JavaInfo.<init>(JavaInfo.java:87)
	... 1 more
```

Open Liberty with the same code and same base Java version (8u202) does not cause issues on AdoptOpenJDK, OpenJ9 and Oracle JDK. I'm unsure if this is an issue in the Java version number parsing in the Open Liberty code or in GraalVM for providing a wrong Java version. I also created this issue here: https://github.com/oracle/graal/issues/1214

**Comment by aguibert** [comment]
hi @MaartenSmeets, thanks for raising this issue. 

This looks like an error attempting to parse the service release of the `java.runtime.version` system property.

I've done a quick run on Graal and here is the system properties dump for future reference:
```
{java.runtime.name=OpenJDK Runtime Environment,
 sun.boot.library.path=/opt/graalvm-ce-1.0.0-rc16/jre/lib/amd64,
 java.vm.version=25.202-b08-jvmci-0.59,
 java.vm.vendor=Oracle Corporation,
 java.vendor.url=http://java.oracle.com/,
 path.separator=:,
 java.vm.name=OpenJDK GraalVM CE 1.0.0-rc16,
 file.encoding.pkg=sun.io,
 user.country=US,
 sun.java.launcher=SUN_STANDARD,
 sun.os.patch.level=unknown,
 java.vm.specification.name=Java Virtual Machine Specification,
 user.dir=/,
 java.runtime.version=1.8.0_202-20190206132807.buildslave.jdk8u-src-tar--b08,
 java.awt.graphicsenv=sun.awt.X11GraphicsEnvironment,
 java.endorsed.dirs=/opt/graalvm-ce-1.0.0-rc16/jre/lib/endorsed,
 os.arch=amd64,
 java.io.tmpdir=/tmp,
 line.separator=
,
 java.vm.specification.vendor=Oracle Corporation,
 os.name=Linux,
 sun.jnu.encoding=UTF-8,
 java.library.path=/opt/rh/llvm-toolset-7/root/usr/lib64:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib,
 java.specification.name=Java Platform API Specification,
 java.class.version=52.0,
 sun.management.compiler=HotSpot 64-Bit Tiered Compilers,
 os.version=4.9.125-linuxkit,
 user.home=/root,
 user.timezone=,
 java.awt.printerjob=sun.print.PSPrinterJob,
 file.encoding=UTF-8,
 java.specification.version=1.8,
 java.class.path=.,
 user.name=root,
 java.vm.specification.version=1.8,
 sun.java.command=TestA,
 java.home=/opt/graalvm-ce-1.0.0-rc16/jre,
 sun.arch.data.model=64,
 user.language=en,
 java.specification.vendor=Oracle Corporation,
 awt.toolkit=sun.awt.X11.XToolkit,
 java.vm.info=mixed mode,
 java.version=1.8.0_202,
 java.ext.dirs=/opt/graalvm-ce-1.0.0-rc16/jre/lib/ext:/usr/java/packages/lib/ext,
 sun.boot.class.path=/opt/graalvm-ce-1.0.0-rc16/jre/lib/resources.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/rt.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/sunrsasign.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/jsse.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/jce.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/charsets.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/jfr.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/jvmci-services.jar:/opt/graalvm-ce-1.0.0-rc16/jre/classes:/opt/graalvm-ce-1.0.0-rc16/jre/lib/boot/graal-sdk.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/boot/graaljs-scriptengine.jar:/opt/graalvm-ce-1.0.0-rc16/jre/lib/boot/graal-sdk.src.zip,
 java.vendor=Oracle Corporation,
 file.separator=/,
 java.vendor.url.bug=http://bugreport.sun.com/bugreport/,
 sun.io.unicode.encoding=UnicodeLittle,
 sun.cpu.endian=little,
 jvmci.Compiler=graal,
 sun.cpu.isalist=}
```

**Comment by aguibert** [comment]
@MaartenSmeets I was able to take a simple hello JAX-RS app and get it running using the Dockerfile and build+run commands you originally indicated. However, when I pinged my app endpoint it all worked fine, and I'm unable to reproduce the issue. This was using `Open Liberty 19.0.0.1 (wlp-1.0.24.cl190120190124-2339)`.

I'll re-try with a more current version of Liberty next. The subject of this issue is a fairly straightforward string parsing issue on our end, so we can put out a fix in a bit.

**Comment by aguibert** [comment]
Was able to reproduce the issue locally using Liberty 19.0.0.4, and then was able to write up a fix and confirm it fixed the issue. Will have it pushed in time for 19.0.0.5.

**Comment by MaartenSmeets** [comment]
Thank you for your quick response and fix! When 19.0.0.5 is available I'll also confirm it has been fixed. This is my first public GitHub issue. If there is anything required of me on the administrative side of things, please let me know.

**Comment by aguibert** [comment]
you're welcome! I've merged the fix now, so it should be available in a nightly build in about 24-48 hours at this location: https://openliberty.io/downloads/#development_builds
Otherwise, it will be included in the next official release (19005)
